ann_benchmarks/algorithms/base.py

```diff
@@ -19,9 +19,12 @@ def query(self, q, n):
         return []  # array of candidate indices
 
     def batch_query(self, X, n):
-        self.res = []
-        for q in X:
-            self.res.append(self.query(q, n))
+        """Provide all queries at once and let algorithm figure out 
+           how to handle it. Default implementation uses a ThreadPool
+           to parallelize query processing."""
+        from multiprocessing.pool import ThreadPool
```

Comment:
super minor but i'd have a preference for this at the top module level

Reply:
Fixed.